packages/react/src/helpers/reads.ts

```diff
@@ -132,18 +156,72 @@ type InferPaginatedItemsType<T extends PaginatedQueryData<unknown>> = T extends
   ? R
   : never;
 
+function useAdHocQuery<
+  TVariables extends PaginatedQueryVariables,
+  TData extends PaginatedQueryData<TItem>,
+  TItem = InferPaginatedItemsType<TData>,
+>(query: DocumentNode): LazyQueryExecFunction<TData, TVariables> {
+  const { apolloClient } = useSharedDependencies();
+  const [fetch] = useLazyQuery<TData, TVariables>(query, {
+    fetchPolicy: 'no-cache',
+    client: apolloClient,
+  });
+
+  return fetch;
+}
+
 export function usePaginatedReadResult<
-  V,
-  T extends PaginatedQueryData<K>,
-  K = InferPaginatedItemsType<T>,
->({ error, data, loading, fetchMore }: ApolloQueryResult<T, V>): PaginatedReadResult<K> {
+  TVariables extends PaginatedQueryVariables,
+  TData extends PaginatedQueryData<TItem>,
+  TItem = InferPaginatedItemsType<TData>,
+>({
+  error,
+  data,
+  loading,
+  fetchMore,
+  variables,
+  observable,
+}: ApolloQueryResult<TData, TVariables>): PaginatedReadResult<TItem[]> {
+  const fetch = useAdHocQuery<TVariables, TData, TItem>(observable.query);
+  const cachedData = useRef(data).current;
+
+  const [beforeCount, setBeforeCount] = useState(0);
+
+  const probeForNewerResults = useCallback(
+    async function (prev: Cursor) {
+      const response = await fetch({
+        variables: {
+          ...variables,
+          cursor: prev,
+        } as TVariables,
+      });
+
+      if (response.data) {
+        setBeforeCount(response.data.result.items.length);
```

Comment:
I don't get how `beforeCount` is set. I tested the new example in `UsePublications` and button never appeared for me when creating a new post from another app.
The `prev` call works as expected, fetching newer posts. but `beforeCount` never changes.

Reply:
ok, after reading the original issue thread (https://github.com/lens-protocol/lens-sdk/issues/419), all clear now, it's good to go 👍